modules/apm/src/test/java/org/elasticsearch/telemetry/apm/internal/metrics/GaugeAdapterTests.java

```diff
@@ -8,116 +8,130 @@
-    Meter testMeter = Mockito.mock(Meter.class);
-    LongGaugeBuilder longGaugeBuilder = Mockito.mock(LongGaugeBuilder.class);
-    DoubleGaugeBuilder mockDoubleGaugeBuilder = Mockito.mock(DoubleGaugeBuilder.class);
+    RecordingMeterProvider meter;
 
     @Before
     public void init() {
-        when(longGaugeBuilder.setDescription(Mockito.anyString())).thenReturn(longGaugeBuilder);
-        when(longGaugeBuilder.setUnit(Mockito.anyString())).thenReturn(longGaugeBuilder);
-
-
-        when(mockDoubleGaugeBuilder.ofLongs()).thenReturn(longGaugeBuilder);
-        when(mockDoubleGaugeBuilder.setUnit(Mockito.anyString())).thenReturn(mockDoubleGaugeBuilder);
-        when(mockDoubleGaugeBuilder.setDescription(Mockito.anyString())).thenReturn(mockDoubleGaugeBuilder);
-        when(testMeter.gaugeBuilder(anyString())).thenReturn(mockDoubleGaugeBuilder);
+        meter = new RecordingMeterProvider();
     }
 
     // testing that a value reported is then used in a callback
     @SuppressWarnings("unchecked")
     public void testLongGaugeRecord() {
-        LongGaugeAdapter longGaugeAdapter = new LongGaugeAdapter(testMeter, "name", "desc", "unit");
+        LongGaugeAdapter gauge = new LongGaugeAdapter(meter, "name", "desc", "unit");
+
+        // recording values;
+        Map<String, Object> m1 = Map.of("k", 1L);
+        Map<String, Object> m3 = Map.of("k", 2L);
+        Map<String, Object> m4 = Map.of("k", 1L, "j", 3L);
+        gauge.record(1L, m1);
+        gauge.record(2L, m1);
+        gauge.record(3L, m3);
+        gauge.record(4L, m4);
+
+        meter.collectMetrics();
+
+        List<TestMetric> metrics = meter.getRecorder().get(gauge);
+
+        assertThat(metrics, hasSize(3));
+
+        assertThat(metrics, containsInAnyOrder(new TestMetric(2L, m1), new TestMetric(3L, m3), new TestMetric(4L, m4)));
 
-        // recording a value
-        longGaugeAdapter.record(1L, Map.of("k", 1L));
+        meter.clearCalls();
 
-        // upon metric export, the consumer will be called
-        ArgumentCaptor<Consumer<ObservableLongMeasurement>> captor = ArgumentCaptor.forClass(Consumer.class);
-        verify(longGaugeBuilder).buildWithCallback(captor.capture());
+        meter.collectMetrics();
+        metrics = meter.getRecorder().get(gauge);
+        assertThat(metrics, empty());
 
-        Consumer<ObservableLongMeasurement> value = captor.getValue();
-        // making sure that a consumer will fetch the value passed down upon recording of a value
-        TestLongMeasurement testLongMeasurement = new TestLongMeasurement();
-        value.accept(testLongMeasurement);
+        gauge.record(1L, m1);
+        gauge.record(3L, m3);
 
-        assertThat(testLongMeasurement.value, Matchers.equalTo(1L));
-        assertThat(testLongMeasurement.attributes, Matchers.equalTo(Attributes.builder().put("k", 1).build()));
+        meter.collectMetrics();
+        metrics = meter.getRecorder().get(gauge);
+        assertThat(metrics, hasSize(2));
```

Comment:
if you check containsInAnyOrder you don't need to check for a size. It would fail if an actual collection has less,more elements then expected matchers